rest_framework_nested/viewsets.py

```diff
@@ -1,24 +1,34 @@
+from __future__ import annotations
+
 import contextlib
+from typing import Any, Generator, Generic, TypeVar, cast
 
 from django.core.exceptions import ImproperlyConfigured
+from django.db.models import Model, QuerySet
+from django.http import HttpRequest, QueryDict
+from rest_framework.generics import GenericAPIView
+from rest_framework.request import Request
+from rest_framework.viewsets import ViewSetMixin
+
+T_Model = TypeVar('T_Model', bound=Model)
 
 
 @contextlib.contextmanager
-def _force_mutable(querydict: dict) -> dict:
+def _force_mutable(querydict: QueryDict | dict[str, Any]) -> Generator[QueryDict | dict[str, Any], None, None]:
```

Comment:
What about reducing this one?
```suggestion
def _force_mutable(querydict: QueryDict | dict[str, Any]) -> Iterable[QueryDict | dict[str, Any]]:
```

From https://docs.python.org/3/library/typing.html#typing.Generator :

> ```
> If your generator will only yield values, set the SendType and ReturnType to None:
> 
> def infinite_stream(start: int) -> Generator[int, None, None]:
>     while True:
>         yield start
>         start += 1
> 
> Alternatively, annotate your generator as having a return type of either Iterable[YieldType] or Iterator[YieldType]:
> 
> def infinite_stream(start: int) -> Iterator[int]:
>     while True:
>         yield start
>         start += 1
> ```